### ISSUE TITLE
Add support for jsx indentation

### ISSUE THREAD

**Issue by LuisCoderDev** [solidjs-community/eslint-plugin-solid#76]
**Describe the need**
i need identation of the html in a component
a example of this exist in this package for react `eslint-plugin-react`, the rule is:

```json
"react/jsx-indent": ["error", 2],
```

this makes identation of hte html return in a component: 

**Example**

Error

```tsx
const App = () =>
{
  return (
    <div>
                                    </div>
  );
};
```

Good

```tsx
const App = () =>
{
  return (
    <div>
    </div>
  );
};
```

**Comment by LuisCoderDev** [comment]
the rule `"solid/self-closing-comp"` dont support indentation, and also there are pleople like me than dont use prettier, eslint, eslint it is also a formarter and should do it 

yes i can put `eslint-plugin-react` in my workspace, but is still incorrect, also i need to reconfigure the rule of eslint-react, beacuse format the code puttin clasname and asking for more tings that solid dont use. for example for now i have this

```json
    "react/jsx-indent": [
      "off",
      2
    ],
    "react/react-in-jsx-scope": "off",
    "react/no-unknown-property": "off",
```

i have to urn off oher properties od eslit react to get it work only the indentation.

**Comment by joshwilsonvu** [comment]
If you don't put `"plugin/react:recommended"` in your ESLint `"extends"` array, you won't have to disable other rules. Just include `"react"` in the `"plugins"` array and only enable the indentation rule.

If you dislike Prettier you may find other code formatters better to work with. But this project is not going to support much in the way of formatting as most people use other solutions.

**Comment by CorentinTh** [comment]
For people that does not want to use prettier ([Why I don't use Prettier](https://antfu.me/posts/why-not-prettier) from antfu)
use the react eslint package as mentionned above:

1. Install `eslint-plugin-react`
```shell
pnpm i -D eslint-plugin-react
```

2. Update your eslint config (add the react plugin, but don't extend)  
```json5
{
    "plugins": [ /* other plugins */, "react"], // <-- add react plugin
    // ...
    "rules": {
        // ...
        "react/jsx-indent": ["error", 2], // <-- add those two rules
        "react/jsx-indent-props": ["error", 2] 
    }
}
```

3. Enjoy
